cmd/fyne_demo/screens/widget.go

```diff
@@ -305,18 +305,29 @@ func makeListTab() fyne.CanvasObject {
 			return len(data)
 		},
 		func() fyne.CanvasObject {
-			return fyne.NewContainerWithLayout(layout.NewHBoxLayout(), widget.NewIcon(theme.DocumentIcon()), widget.NewLabel("Template Object"))
+			return fyne.NewContainerWithLayout(layout.NewHBoxLayout(), widget.NewIcon(theme.DocumentIcon()), layout.NewSpacer(), widget.NewLabel("Template Object"))
 		},
 		func(index int, item fyne.CanvasObject) {
-			item.(*fyne.Container).Objects[1].(*widget.Label).SetText(data[index])
+			item.(*fyne.Container).Objects[2].(*widget.Label).SetText(data[index])
 		},
 	)
 	list.OnItemSelected = func(index int) {
 		label.SetText(data[index])
 		icon.SetResource(theme.DocumentIcon())
 	}
+	clearButton := widget.NewButton("Clear Contents",
```

Comment:
It seems like this is really deomnstration that a bug is fixed - so it should be in a unit test, not the demo UI...

Reply:
I added more unit tests should be good to go now

Reply:
The new tests are great thanks, but do we need this row of buttons in the demo?

Reply:
I'm not sure.  I think it's nice to exhibit that the list is dynamic but I guess that is the teams call.  I just put it there to show how list could add/remove items dynamically.  (And it is nice to have for when people say okay to replicate add some list items, then remove one, then clear, then add, then remove, boom crash, etc...)

Reply:
It's just not how any of the other widgets are presented, so seems strange in comparison